### PR TITLE
Runs swift 4.2 migration tool and gets project compiling again

### PR DESCRIPTION
**Pull request by simonmitchell**

### CODE REVIEW

File: ThunderCloud/AppCollectionCell.swift [hunk 1]
```diff
@@ -103,7 +104,7 @@ extension AppCollectionCell {
 				handler: { (action) in
 					
 					NotificationCenter.default.sendStatEventNotification(category: "Collect them all", action: "Open", label: nil, value: nil, object: self)
-					UIApplication.shared.open(launchURL, options: [:], completionHandler: nil)
+					UIApplication.shared.open(launchURL, options: convertToUIApplicationOpenExternalURLOptionsKeyDictionary([:]), completionHandler: nil)
```
Comment (MattCheetham) [inline ThunderCloud/AppCollectionCell.swift]: Take the options param out as it isn't used. Then scrap the convert method

File: ThunderCloud/DeveloperModeController.swift [hunk 2]
```diff
@@ -199,7 +199,7 @@ public class DeveloperModeController: NSObject {
                 
             uiWindow = UIWindow(frame: UIScreen.main.bounds)
             uiWindow?.rootViewController = loginViewController
-            uiWindow?.windowLevel = UIWindowLevelAlert+1
+            uiWindow?.windowLevel = UIWindow.Level.alert+1
```
Comment (MattCheetham) [inline ThunderCloud/DeveloperModeController.swift]: I believe `.alert + 1` will suffice

File: ThunderCloud/PokemonListItemView.swift [hunk 3]
```diff
@@ -140,7 +140,7 @@ extension PokemonListItemView: PokemonTableViewCellDelegate {
 					handler: { (action) in
 						
 						NotificationCenter.default.sendStatEventNotification(category: "Collect them all", action: "Open", label: nil, value: nil, object: self)
-						UIApplication.shared.open(localLink, options: [:], completionHandler: nil)
+						UIApplication.shared.open(localLink, options: convertToUIApplicationOpenExternalURLOptionsKeyDictionary([:]), completionHandler: nil)
```
Comment (MattCheetham) [inline ThunderCloud/PokemonListItemView.swift]: Again, remove here

File: ThunderCloud/QuizTextSelectionViewController.swift [hunk 4]
```diff
@@ -24,7 +24,7 @@ class QuizTextSelectionViewController: TableViewController {
 		guard let question = question else { return }
 		
 		tableView.allowsMultipleSelection = question.limit > 1
-		tableView.contentInset = UIEdgeInsetsMake(0, 0, 44+32, 0)
+		tableView.contentInset = UIEdgeInsets.init(top: 0, left: 0, bottom: 44+32, right: 0)
```
Comment (MattCheetham) [inline ThunderCloud/QuizTextSelectionViewController.swift]: Do you need `.init`?

Comment (simonmitchell) [reply ThunderCloud/QuizTextSelectionViewController.swift]: Nope, definitely not! Swift migration tool -.-

File: ThunderCloud/PokemonListItemView.swift (hunk 3 above)
Comment (simonmitchell) [reply ThunderCloud/PokemonListItemView.swift]: I hate the 4.2 migration tool 😞

File: ThunderCloud/AppCollectionCell.swift (hunk 1 above)
Comment (simonmitchell) [reply ThunderCloud/AppCollectionCell.swift]: Eurghhh, stupid migration tool!

**Pull request merged**